pyk/src/pyk/integration_tests/kore/test_kore_client.py

```diff
@@ -0,0 +1,54 @@
+from string import Template
+from typing import Any, Final, Mapping, Tuple
+
+from pyk.kore.client import BranchingResult, DepthBoundResult, ExecuteResult, State, StuckResult
+from pyk.kore.parser import KoreParser
+from pyk.kore.syntax import Pattern
+
+from .kore_client_test import KoreClientTest
+
+
+def term(n: int) -> Pattern:
+    template = Template(
+        '''
+    Lbl'-LT-'generatedTop'-GT-'{}(
+        Lbl'-LT-'k'-GT-'{}(
+            kseq{}(
+                inj{SortInt{}, SortKItem{}}(
+                    \dv{SortInt{}}("$n")
+                ),
+                K:SortK{}
+            )
+        ),
+        GCC:SortGeneratedCounterCell{}
+    )
+    '''
+    )
+    parser = KoreParser(template.substitute(n=n))
+    pattern = parser.pattern()
+    assert parser.eof
+    return pattern
+
+
+def state(n: int) -> State:
+    return State(term=term(n), substitution=None, predicate=None)
+
+
+class SimpleKoreClientTest(KoreClientTest):
+    KOMPILE_MAIN_FILE = 'k-files/kore-rpc-test.k'
+    KORE_MODULE_NAME = 'KORE-RPC-TEST'
+
+    TEST_DATA: Final[Tuple[Tuple[int, Mapping[str, Any], ExecuteResult], ...]] = (
+        (0, {}, BranchingResult(state=state(2), depth=2, next_states=(state(4), state(3)))),
+        (0, {'max_depth': 2}, DepthBoundResult(state=state(2), depth=2)),
+        (4, {}, StuckResult(state=state(6), depth=3)),
```

Comment:
Why is this reporting back depth 3? Sholudn't it be depth 2? 4 => 5 => 6 is two rewrite steps.

Reply:
@jberthold, is this on purpose or a bug in the backed?

Reply:
I think it is probably a bug, resulting from the logic that implements the steps.
When we are in state `6` here, the execution does perform a "rewrite step", but which results in state `Remaining 6` (could not be rewritten). However, the "remaining state" would still be simplified so it is not necessarily the same as before (it is here, of course). 
Will analyse this example to confirm that, and fix it. I think calling the simplifier again is redundant here anyway but have to check.

Reply:
Issue made: https://github.com/runtimeverification/haskell-backend/issues/3210